Setup_audio

monke6942021 opened the pull request:
Croissant can now recognize sc:AudioObject types, but can't pull any relevant data.

marcenacp commented on requirements.txt:
```diff
@@ -0,0 +1,27 @@
+absl-py==1.4.0
```
We don't use requirements.txt, but [`pyproject.toml`](https://github.com/mlcommons/croissant/blob/main/python/mlcroissant/pyproject.toml) to manage dependencies.

marcenacp commented on python/mlcroissant/mlcroissant/_src/core/data_types.py:
```diff
@@ -28,6 +28,9 @@ def check_expected_type(issues: Issues, jsonld: Json, expected_type: str):
     constants.SCHEMA_ORG_DATA_TYPE_IMAGE_OBJECT: (
         constants.SCHEMA_ORG_DATA_TYPE_IMAGE_OBJECT
     ),
+    constants.SCHEMA_ORG_DATA_TYPE_AUDIO_OBJECT: (
```
I believe this has changed. Can you rebase to have a freshest version of the code?

```
git pull --rebase origin main
```

marcenacp reviewed (changes requested):
Thanks! Do you have a dataset to test this?

monke6942021 commented:
> Thanks! Do you have a dataset to test this?

I just added the one that I've been using for testing. It's a mini-version of what I finally want to do with croissant, but maybe the 0-hz-immersion-into-wallbient directory can be used as another test if we want it to operate without looking into multiple directories.

marcenacp reviewed (changes requested):
Thanks for the dataset. My remarks:

- The dataset should probably be added to the [non-hermetic unit tests](https://github.com/mlcommons/croissant/blob/main/python/mlcroissant/mlcroissant/_src/datasets_test.py#L111). Otherwise, we'll have regression for that.
- Is it possible to not commit the *.mp3 and rather host them somewhere? Otherwise, it could make the repo heavier.
- Could you address the previous reviews? (E.g., rebase your code).

Thanks!

monke6942021 commented:
Hi! I added the dataset to the non-hermetic tests, and I'm also hosting the dataset from a Google Drive, which I shared with you and Tim. Also, sorry about the problem with the rebase. I thought I rebased it when I added the dataset, but I did not complete the process. I have now rebased the branch and synced the changes. I hope it works properly now. I also removed the requirements.txt file.

monke6942021 commented:
I passed the tests. Is there anything else I should do?

marcenacp commented on python/mlcroissant/mlcroissant/_src/datasets_test.py:
```diff
@@ -127,6 +127,11 @@ def test_hermetic_loading(dataset_name, record_set_name, num_records):
+            "audio_test/metadata.json",
+            "records",
+            -1,
+        ],  # Switch the number to 10 if nessacary
```
necessary

marcenacp replied:
Remove the comment?

monke6942021 replied:
I just removed it.

marcenacp commented on python/mlcroissant/mlcroissant/_src/core/optional.py:
```diff
@@ -86,5 +86,10 @@ def PIL_Image(cls) -> types.ModuleType:  # pylint: disable=invalid-name
 
+    @cached_class_property
+    def LIB_Audio(cls) -> types.ModuleType:  # pylint: disable=invalid-name
+        """Cached git module."""
```
`Cached librosa module`

marcenacp commented on python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:
```diff
@@ -31,6 +31,9 @@ def _cast_value(value: Any, data_type: type | term.URIRef | None):
             raise ValueError(f"Type {type(value)} is not accepted for an image.")
+    elif data_type == DataType.AUDIO_OBJECT:
+        output = deps.LIB_Audio.load(io.BytesIO(value))
+        return str([output[0].tolist(), output[1]])
```
Why a str? How will you use this in an ML pipeline? What would be the most useful signal to return here?

monke6942021 replied:
I did it like that because the output needs to have double quotes. I'm about to push a version that just outputs the regular librosa output, but I don't know if it will pass the test.

marcenacp commented on python/mlcroissant/pyproject.toml:
```diff
@@ -18,6 +19,7 @@ authors = [
 # pip dependencies of the project
 # Installed locally with `pip install -e .`
 dependencies = [
+  "black[jupyter]",
```
Remove jupyter?

marcenacp commented on python/mlcroissant/mlcroissant/_src/core/optional.py:
```diff
@@ -86,5 +86,10 @@ def PIL_Image(cls) -> types.ModuleType:  # pylint: disable=invalid-name
         return _try_import("PIL.Image", package_name="Pillow")
 
+    @cached_class_property
+    def LIB_Audio(cls) -> types.ModuleType:  # pylint: disable=invalid-name
```
`def librosa` (PIL.Image above is a library, so the name here should probably be librosa)

monke6942021 replied:
I just changed the name, and the usage in field.py
